@pdettori It's on the test side, I believe. Please take a look at the output.

```
=========================== short test summary info ============================
FAILED tests/e2e/common/test_agent_conversation.py::TestWeatherAgentConversation::test_agent_simple_query - Failed: Agent returned a FAILED task
  Agent URL: http://localhost:8000
  Query: What is the weather like in San Francisco?
  Error: Error: No LLM API key configured. Set the LLM_API_KEY environment variable.
  Task details:
    task.status.state=TaskState.failed
    task.artifacts count=1
      artifact[0] text='Error: No LLM API key configured. Set the LLM_API_KEY environment variable.'
====== 1 failed, 36 passed, 18 skipped, 34 deselected in 62.15s (0:01:02) ======
```

The test expects an LLM API key, but it's not provided. It's definitely not in my code.

--- a/kagenti/backend/app/routers/agents.py
+++ b/kagenti/backend/app/routers/agents.py
@@ -2004,6 +2004,40 @@ def _ensure_authbridge_scc_rolebinding(
     )
 
 
+def _ensure_card_unsigned_configmap(
+    kube: KubernetesService,
+    name: str,
+    namespace: str,
+    service_port: int = DEFAULT_IN_CLUSTER_PORT,
+) -> None:
+    """Create the <agent>-card-unsigned ConfigMap if it does not exist.
+
+    The Kagenti operator webhook checks for this ConfigMap when a
+    Deployment is admitted.  If it exists, the webhook injects a
+    ``sign-agentcard`` init container that signs the agent card with
+    the workload's SPIRE SVID.  The ConfigMap must therefore be
+    created **before** the Deployment.
+    """
+    agent_url = f"http://{name}.{namespace}.svc.cluster.local:{service_port}"
+    agent_card = json.dumps(
+        {
+            "name": name,
+            "url": agent_url,
+            "version": "1.0.0",
+            "capabilities": {},
+            "defaultInputModes": ["application/json"],
+            "defaultOutputModes": ["text/plain"],
+            "skills": [],
+        },
+        indent=2,
+    )
+    kube.ensure_configmap(
+        namespace=namespace,
+        name=f"{name}-card-unsigned",
+        data={"agent.json": agent_card},
+    )
+
+
 def _build_agent_shipwright_build_manifest(
     request: CreateAgentRequest, clone_secret_name: Optional[str] = None
 ) -> dict:
@@ -2687,6 +2721,21 @@ async def create_agent(
             if request.authBridgeEnabled:
                 _ensure_authbridge_scc_rolebinding(kube=kube, namespace=request.namespace)
 
+            # Create card-unsigned ConfigMap so the webhook injects
+            # the sign-agentcard init container at Deployment admission.
+            if request.spireEnabled:
+                service_port = (
+                    request.servicePorts[0].port
+                    if request.servicePorts
+                    else DEFAULT_IN_CLUSTER_PORT
+                )
+                _ensure_card_unsigned_configmap(
+                    kube=kube,
+                    name=request.name,
+                    namespace=request.namespace,
+                    service_port=service_port,
+                )
+
             # Create workload based on workloadType
             if request.workloadType == WORKLOAD_TYPE_DEPLOYMENT:
                 workload_manifest = _build_deployment_manifest(
@@ -3156,6 +3205,19 @@ async def finalize_shipwright_build(
         # On OpenShift, ensure the AuthBridge SCC RoleBinding exists
         if final_auth_bridge:
             _ensure_authbridge_scc_rolebinding(kube=kube, namespace=namespace)
+
+        # Create card-unsigned ConfigMap so the webhook injects
+        # the sign-agentcard init container at Deployment admission.
+        if final_spire_enabled:
+            service_port = (
+                final_service_ports[0].port if final_service_ports else DEFAULT_IN_CLUSTER_PORT
+            )
+            _ensure_card_unsigned_configmap(
+                kube=kube,
+                name=name,
+                namespace=namespace,
+                service_port=service_port,
+            )
 
         # Create workload based on workloadType
         if final_workload_type == WORKLOAD_TYPE_DEPLOYMENT:

